go/merkletree/merkle_verifier.go

```diff
@@ -47,15 +47,16 @@ func (m MerkleVerifier) VerifyInclusionProof(leafIndex, treeSize int64, proof []
 }
 
 // RootFromInclusionProof calculates the expected tree root given the proof and leaf.
+// leafIndex starts at 0. treeSize starts at 1.
 func (m MerkleVerifier) RootFromInclusionProof(leafIndex, treeSize int64, proof [][]byte, leaf []byte) ([]byte, error) {
-	if leafIndex > treeSize {
+	if leafIndex >= treeSize {
 		return nil, fmt.Errorf("leafIndex %d > treeSize %d", leafIndex, treeSize)
 	}
-	if leafIndex == 0 {
-		return nil, errors.New("leafIndex is zero")
+	if leafIndex < 0 || treeSize < 1 {
+		return nil, errors.New("leafIndex < 0 or treeSize < 1")
 	}
 
-	node := leafIndex - 1
+	nodeIndex := leafIndex
```

Comment:
And good call on renaming this. I know the Go folks would probably just call this `n` or `i`, but this code is tricky enough as it is... :stuck_out_tongue: